retry.go

```diff
@@ -0,0 +1,105 @@
+package aviation
+
+import (
+	"context"
+	"time"
+
+	"github.com/jpillora/backoff"
+	"github.com/mongodb/grip"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+func MakeRetryUnaryClientInterceptor(maxRetries int) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		var lastErr error
+		b := &backoff.Backoff{
+			// TODO: figure out the best options for this.
+			Min:    100 * time.Millisecond,
+			Max:    time.Second,
+			Factor: 2,
+		}
+		callCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+	retry:
+		for i := 0; i < maxRetries; i++ {
+			lastErr = invoker(callCtx, method, req, reply, cc, opts...)
+			grip.Infof("GRPC client retry attempt: %d, error: %v", i, lastErr)
+			if lastErr == nil {
+				return nil
+			}
+
+			if !isRetriable(lastErr) {
+				break
+			}
+
+			timer := time.NewTimer(b.Duration())
+			select {
+			case <-ctx.Done():
+				break retry
+			case <-timer.C:
+				continue retry
+			}
+		}
+
+		grip.Warning("GRPC client retries exceeded or canceled!")
```

Comment:
use a message.fields and say how long the request took/how many retries. 

it'd also be nice if we could get the request url/method name to provide additional context.